### DISCUSSION TITLE
Why MetaMask always asking 2 transactions FundMe.sol

### DISCUSSION THREAD

**Discussion by helalmerchan** [Cyfrin/foundry-full-course-cu#2007] [General]
I Just deployed the FundMe.sol on SepoliaETH 

All functions working well on remix **except**  asking 2 transactions when I call the funding button/function.

[Latest transactions](https://sepolia.etherscan.io/address/0xca131c6ca68bb5b963617a3e3ed32bb65dcb17e8)

here is my remix code

```solidity
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.18;

// solhint-disable-next-line interface-starts-with-i
import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol";

contract FundMe{
    uint256 public minimumUsd = 5e18;

    address[] public founders;
    
    function funding() public payable{
        require(getConvertionRate(msg.value) >= minimumUsd, "Minimum 5 usd to participate");
        founders.push(msg.sender);
    }
    // function withdraw() public{}

    function getPrice() public view returns(uint256){
        // address 0x694AA1769357215DE4FAC081bf1f309aDC325306
        // ABI  
        AggregatorV3Interface priceFeed = AggregatorV3Interface(0x694AA1769357215DE4FAC081bf1f309aDC325306);
        (,int256 price,,,) =  priceFeed.latestRoundData();   
        return uint256(price * 1e10);
    }

    function getConvertionRate(uint256 ethAmount) public view returns(uint256){
        uint256 ethPrice = getPrice();
        uint256 ethAmountInUsd = (ethPrice * ethAmount) / 1e18;
        return ethAmountInUsd;
    }

    function getVersion() public view returns(uint256){
        return AggregatorV3Interface(0x694AA1769357215DE4FAC081bf1f309aDC325306).version();
    }
}

```

**Comment by EngrPips** [comment]
Hello @helalmerchan, I checked the etherscan link you sent, and every transaction sent to the contract appeared to be successful, including the `fund` functions. I didn't see any issue, or am I not understanding properly?

**Reply by helalmerchan** [reply]
Hello @EngrPips thanks for your reply. My question is why Metamask asking for 2 transactions in a single call. You'll see 2 transactions in same timestamp in my given transaction link. It is charging me twice my fund :(

**Reply by EngrPips** [reply]
Oh, I see. The `funding` transaction modifies the state of the blockchain, so you first need to sign the transaction and then send it. That is what I assume is happening, but maybe you can ask an AI buddy and see what it says.
